sdk/src/invariant.ts

```diff
@@ -522,6 +527,72 @@ export class Invariant {
   }
 
   async getAllPositions(
+    owner: string,
+    positionsCount?: bigint,
+    skipPages?: number[],
+    options: ContractOptions = {
+      storageDepositLimit: this.storageDepositLimit,
+      refTime: this.gasLimit.refTime.toNumber(),
+      proofSize: this.gasLimit.proofSize.toNumber()
+    }
+  ): Promise<Page[]> {
+    let firstPageIndex = 0
+    for (let i = 1; i < Number.MAX_SAFE_INTEGER; i++) {
```

Comment:
More in this direction, it scales significantly better:
```Typescript
firstPageIndex = skipPages?.find(i => !skipPages.includes(i)) || 1;
```

btw try to ask chatGPT more often about possible refactorings